src/tests/ftest/pool/evict_test.py

```diff
@@ -38,110 +39,257 @@ class EvictTest(TestWithServers):
+        bogus_uuid = '81ef94d7-a59d-4a5e-935b-abfbd12f2105'
+        # set the UUID directly
+        self.pool.pool.set_uuid_str(bogus_uuid)
+        self.log.info("Evicting pool with Invalid Pool UUID:  %s",
+                      self.pool.pool.get_uuid_str())
         try:
-            # use the wrong server group name but the correct uuid
-            bogus_server_group = self.params.get("setname",
-                                                 '/run/tests/setnames/'
-                                                 'badsetname/*')
-            bad_evict_cmd = (
-                "{0} evict-pool "
-                "-i {1} "
-                "-s {2}".format(self.daosctl, uuid_str, bogus_server_group))
-            process.system(bad_evict_cmd)
-
-        except Exception as excep:
-            # this section of the test should fail and throw an exception
-            print("command expected to fail")
+            self.log.info("Attempting to evict pool with an invalid UUID")
+            # call daos evict api directly
+            self.pool.pool.evict()
+        # exception is expected
+        except DaosApiError as result:
+            if "-1005" in str(result):
+                self.log.info(
+                    "Expected exception: evicting pool with invalid UUID\n %s",
+                    str(result))
+            else:
+                self.log.info(
+                    "Unexpected exception: \n %s", str(result))
+            # restore the valid UUID and check if pool handle still exists
+            self.pool.pool.set_uuid_str(saved_uuid)
+            self.log.info("Check if pool handle still exist")
+            self.assertTrue(
+                (int(self.pool.pool.handle.value) > 0),
+                "Pool handle was removed when doing an evict with a bad UUID")
+            return
+        # if here then pool-evict did not raise an DAOS exception as expected
+        # restore the valid UUID and check if valid pool still exists
+        self.log.info(
+            "DAOS api exception did not occur"
+            " - evict pool with invalid UUID")
+        self.pool.pool.set_uuid_str(saved_uuid)
+        self.log.info("check if pool data and pool handle still exist")
+        self.assertTrue(
+            (int(self.pool.pool.handle.value) > 0),
+            "Pool handle was removed when doing an evict with a bad UUID")
+        self.assertTrue(
+            self.pool.check_files(self.hostlist_servers),
+            "Valid pool data was not detected on servers after "
+            "a pool evict with invalid UUID failed to raise an exception")
+        self.fail(
+            "Test did not raise an exception when "
+            "evicting a pool with invalid UUID")
+
+    def test_evict_bad_server_group(self):
+        """
+        Test evicting a pool using an invalid server group name.
+
+        :avocado: tags=pool,poolevict
+        """
+        setid = self.params.get("setname", '/run/setnames/validsetname/')
```

Comment:
This test and test_evict_bad_uuid share a bunch of code and should probably call a common method.